### PR TITLE
Fix/read xml attributes

### PR DESCRIPTION
**Pull request by deichmab-draeger**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have updated the [changelog](../CHANGELOG.md) accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/sdc11073/xml_types/xml_structure.py [hunk 1]
```diff
@@ -477,53 +548,69 @@ def update_xml_value(self, instance, node):
```
Comment (leon1995) [inline src/sdc11073/xml_types/xml_structure.py:548]: line 546: `Unresolved attribute reference 'elem_to_xml' for class 'DataConverterProtocol' `

File: src/sdc11073/xml_types/xml_structure.py [hunk 2]
```diff
@@ -782,22 +895,26 @@ def update_xml_value(self, instance, node):
                 sub_node = self._get_element_by_child_name(node, self._sub_element_name, create_missing_nodes=True)
```
Comment (leon1995) [inline src/sdc11073/xml_types/xml_structure.py:895]: Local variable 'sub_node' value is not used

File: src/sdc11073/xml_types/xml_structure.py [hunk 3]
```diff
@@ -828,15 +946,28 @@ def update_xml_value(self, instance, node):
 
 
 class ContainerProperty(_ElementBase):
-    """ a value that has "mk_node" and cls.from_node methods"""
-
-    def __init__(self, sub_element_name, value_class, cls_getter, ns_helper, is_optional=False):
+    """ContainerProperty is a property that supports xsi:type information from xml and instantiates value accordingly."""
```
Comment (leon1995) [inline src/sdc11073/xml_types/xml_structure.py:949]: line too long

File: src/sdc11073/xml_types/xml_structure.py [hunk 4]
```diff
@@ -828,15 +946,28 @@ def update_xml_value(self, instance, node):
...
+                 value_class: Any,
+                 cls_getter: Callable,
+                 ns_helper: NamespaceHelper,
+                 is_optional: bool = False):
+        """Construct a ContainerProperty.
+
+        :param sub_element_name: see doc of base class
+        :param value_class: Default value class if no xsi:type is found
+        :param cls_getter: function that returns a class for xsi:type QName
+        :param ns_helper: name space helper thatknows current prefixes
```
Comment (leon1995) [inline src/sdc11073/xml_types/xml_structure.py:961]: missing whitespace in `thatknows`

File: src/sdc11073/xml_types/xml_structure.py [hunk 5]
```diff
@@ -914,40 +1043,52 @@ def get_py_value_from_node(self, instance, node):
...
+                 value_class: Any,
+                 cls_getter: Callable,
+                 ns_helper: NamespaceHelper,
+                 is_optional: bool = True):
+        """Construct a list of Containers.
+
+        :param sub_element_name: see doc of base class
+        :param value_class: Default value class if no xsi:type is found
+        :param cls_getter: function that returns a class for xsi:type QName
+        :param ns_helper: name space helper thatknows current prefixes
```
Comment (leon1995) [inline src/sdc11073/xml_types/xml_structure.py:1082]: missing whitespace in `thatknows`

File: src/sdc11073/xml_types/xml_structure.py [hunk 6]
```diff
@@ -828,15 +946,28 @@ def update_xml_value(self, instance, node):
...
 
 class ContainerProperty(_ElementBase):
-    """ a value that has "mk_node" and cls.from_node methods"""
-
-    def __init__(self, sub_element_name, value_class, cls_getter, ns_helper, is_optional=False):
+    """ContainerProperty is a property that supports xsi:type information from xml and instantiates value accordingly."""
+
+    def __init__(self, sub_element_name: etree_.QName | None,  # Noqa: PLR0913
+                 value_class: Any,
+                 cls_getter: Callable,
```
Comment (leon1995) [inline src/sdc11073/xml_types/xml_structure.py:953]: type could be extended to (based on documentation): `Callable[[etree_.QName], typing.Type]`

File: src/sdc11073/xml_types/xml_structure.py [hunk 7]
```diff
@@ -914,40 +1043,52 @@ def get_py_value_from_node(self, instance, node):
...
-    Used if maxOccurs="Unbounded" in BICEPS_ParticipantModel"""
+    """ContainerListProperty is a property with a list of elements, each supports xsi:type information.
+
+    Used if maxOccurs="Unbounded" in BICEPS_ParticipantModel.
+    """
 
-    def __init__(self, sub_element_name, value_class, cls_getter, ns_helper, is_optional=True):
+    def __init__(self, sub_element_name: etree_.QName | None,  # Noqa: PLR0913
+                 value_class: Any,
+                 cls_getter: Callable,
```
Comment (leon1995) [inline src/sdc11073/xml_types/xml_structure.py:1074]: type could be extended to (based on documentation): `Callable[[etree_.QName], typing.Type]`

File: src/sdc11073/xml_types/xml_structure.py [hunk 8]
```diff
@@ -1224,11 +1383,15 @@ class DateOfBirthProperty(_ElementBase):
     The corresponding Python types are datetime.Date (=> not time point available) or datetime.Datetime (with time point attribute)
```
Comment (leon1995) [inline src/sdc11073/xml_types/xml_structure.py:1383]: line too long

**Review by leon1995 — CHANGES_REQUESTED** [review]

File: src/sdc11073/xml_types/xml_structure.py (hunk 1 above)
Comment (deichmab-draeger) [reply src/sdc11073/xml_types/xml_structure.py:548]: I think this cannot be expressed with typing. The dataconverter is a ListConverter in __init__,  which has this method.

**Review by leon1995 — APPROVED** [review]

**Pull request merged**